scripts/ci/run-monitor-ci-tests.bash

```diff
@@ -0,0 +1,224 @@
+# This script uses the CircleCI APIs to make this magic happen.
+#
+# If the private CI fails, this script will collect the names and artifacts of the failed jobs and report them.
+# This script should support multiple workflows if more are added, although it has not been tested.
+# This script waits 40 minutes for the private CI to complete otherwise it fails.
+#
+# **For Running from the UI Repository:**
+# If you retry a failing job in the private CI and it passes, you can safely rerun this job in the public CI.
+#  - This script uses your commit SHA to search for a passing pipeline before starting a new one.
+#  - If you rerun the private CI and it passes, this script will find that pipeline and will not start a new one.
+#  - In this situation the script will exit quickly with success.
+#
+# Required Env Vars for Running from the UI Repository:
+# - SHA: the UI repo commit SHA we're running against
+# - API_KEY: the CircleCI API access key
+# - UI_BRANCH: the branch of the UI repo we're running against
+# - MONITOR_CI_BRANCH: the branch of the monitor-ci repo to start a pipeline with (usually 'master')
+# - PULL_REQUEST: the open pull request, if one exists (used for lighthouse)
+#
+# **For running from the influxdb OSS repository:**
+# Since the OSS private CI is very simple, retrying a failing job in the private CI is not supported.
+# Currently, only running e2e tests against the latest from the UI master branch is supported. 
+#
+# Required Env Vars for Running from the influxdb OSS Repository:
+# - OSS_SHA: the influxdb repo commit SHA we're running against
+# - MONITOR_CI_BRANCH: the branch of the monitor-ci repo to start a pipeline with (usually 'master')
+########################
+
+# make dir for artifacts
+mkdir -p monitor-ci/test-artifacts/results/{build-oss-image,oss-e2e,build-image,cloud-e2e,cloud-e2e-firefox,cloud-e2e-k8s-idpe,cloud-lighthouse,smoke,build-prod-image,deploy}/{shared,oss,cloud}
+# if we are not running from the OSS repo, OSS_SHA will be unset, and we must be running from the UI repo.
+if [[ -z "${OSS_SHA:-}" ]]; then
+	# get monitor-ci pipelines we've already run on this SHA
+	found_passing_pipeline=0
+	all_pipelines=$(curl -s --request GET \
+			--url "https://circleci.com/api/v2/project/gh/influxdata/monitor-ci/pipeline" \
+			--header "Circle-Token: ${API_KEY}" \
+			--header 'content-type: application/json' \
+			--header 'Accept: application/json')
+
+	# check the status of the workflows for each of these pipelines
+	all_pipelines_ids=( $(echo ${all_pipelines} | jq -r '.items | .[].id') )
+	for pipeline_id in "${all_pipelines_ids[@]}"; do
+
+		config=$(curl -s --request GET \
+			--url "https://circleci.com/api/v2/pipeline/${pipeline_id}/config" \
+			--header "Circle-Token: ${API_KEY}" \
+			--header 'content-type: application/json' \
+			--header 'Accept: application/json')
+
+		# finds the UI SHA parameter used in this pipeline by hunting for the line "export UI_SHA="
```

Comment:
This line could do with a bit more explanation of the line formatting of the `.compiled` config line that it expects, i.e. with an example of what it looks like. Otherwise, in case the formatting changes in the future, this line could be difficult to debug.